### PR TITLE
[API/Librus] Fix LibrusApiTeacherFreeDays.

### PR DESCRIPTION
**Pull request by BxOxSxS**
#121

### CODE REVIEW

File: app/src/main/java/pl/szczodrzynski/edziennik/data/api/edziennik/librus/data/api/LibrusApiTeacherFreeDays.kt [hunk 1]
```diff
@@ -36,12 +36,13 @@ class LibrusApiTeacherFreeDays(override val data: DataLibrus,
                 val id = teacherAbsence.getLong("Id") ?: return@forEach
                 val teacherId = teacherAbsence.getJsonObject("Teacher")?.getLong("Id")
                         ?: return@forEach
-                val type = teacherAbsence.getJsonObject("Type").getLong("Id") ?: return@forEach
+                val type = teacherAbsence.getJsonObject("Teacher")?.getLong("Id")
+                    ?: return@forEach
```
Comment (kuba2k2) [inline app/src/main/java/pl/szczodrzynski/edziennik/data/api/edziennik/librus/data/api/LibrusApiTeacherFreeDays.kt:40]: What is this?

**Review by kuba2k2 — CHANGES_REQUESTED** [review]

File: app/src/main/java/pl/szczodrzynski/edziennik/data/api/edziennik/librus/data/api/LibrusApiTeacherFreeDays.kt [hunk 2]
```diff
@@ -36,12 +36,13 @@ class LibrusApiTeacherFreeDays(override val data: DataLibrus,
...
                 val name = data.teacherAbsenceTypes.singleOrNull { it.id == type }?.name
                 val dateFrom = Date.fromY_m_d(teacherAbsence.getString("DateFrom"))
                 val dateTo = Date.fromY_m_d(teacherAbsence.getString("DateTo"))
-                val timeFrom = teacherAbsence.getString("TimeFrom")?.let { Time.fromH_m_s(it) }
-                val timeTo = teacherAbsence.getString("TimeTo")?.let { Time.fromH_m_s(it) }
+                val timeFrom = teacherAbsence.getString("DateFrom")?.let { Time.fromH_m_s(it) }
+                val timeTo = teacherAbsence.getString("DateTo")?.let { Time.fromH_m_s(it) }
```
Comment (kuba2k2) [inline app/src/main/java/pl/szczodrzynski/edziennik/data/api/edziennik/librus/data/api/LibrusApiTeacherFreeDays.kt:45]: Why is this changed? Date is already retrieved above.

**Review by kuba2k2 — CHANGES_REQUESTED** [review]

File: app/src/main/java/pl/szczodrzynski/edziennik/data/api/edziennik/librus/data/api/LibrusApiTeacherFreeDays.kt (hunk 2 above)
Comment (BxOxSxS) [reply app/src/main/java/pl/szczodrzynski/edziennik/data/api/edziennik/librus/data/api/LibrusApiTeacherFreeDays.kt:45]: TimeFrom and TimeTo are not returned in JSON anymore. App could not display teacher absence when any variable was assigned with non existing object. I could not find way to not declarete the data (leak of knowedge and expirence) so I used this one. Yes it's dirty. What is better solution?

File: app/src/main/java/pl/szczodrzynski/edziennik/data/api/edziennik/librus/data/api/LibrusApiTeacherFreeDays.kt (hunk 1 above)
Comment (BxOxSxS) [reply app/src/main/java/pl/szczodrzynski/edziennik/data/api/edziennik/librus/data/api/LibrusApiTeacherFreeDays.kt:40]: Type object is not returned in JSON anymore. App could not display teacher absence when any variable was assigned with non existing object. I could not find way to not declarete the data (leak of knowedge and expirence) so I used this one. Yes it's dirty. What is better solution?

File: app/src/main/java/pl/szczodrzynski/edziennik/data/api/edziennik/librus/data/api/LibrusApiTeacherFreeDays.kt (hunk 2 above)
Comment (kuba2k2) [reply app/src/main/java/pl/szczodrzynski/edziennik/data/api/edziennik/librus/data/api/LibrusApiTeacherFreeDays.kt:45]: If you inspect the resulting object, you will see the time being probably null anyway. Time class cannot parse dates, so it probably simply returns null.

File: app/src/main/java/pl/szczodrzynski/edziennik/data/api/edziennik/librus/data/api/LibrusApiTeacherFreeDays.kt (hunk 1 above)
Comment (kuba2k2) [reply app/src/main/java/pl/szczodrzynski/edziennik/data/api/edziennik/librus/data/api/LibrusApiTeacherFreeDays.kt:40]: The app expects a Type ID in this field. Assigning a Teacher ID does not make it find a "valid" object, because Teacher is not a Type. Try putting -1 as the type ID to see if it works. If not, you have to modify the logic of displaying the absence list in the UI.

File: app/src/main/java/pl/szczodrzynski/edziennik/data/api/edziennik/librus/data/api/LibrusApiTeacherFreeDays.kt [hunk 3]
```diff
@@ -36,12 +36,12 @@ class LibrusApiTeacherFreeDays(override val data: DataLibrus,
...
                         ?: return@forEach
-                val type = teacherAbsence.getJsonObject("Type").getLong("Id") ?: return@forEach
+                val type = "-1".toLong()
                 val name = data.teacherAbsenceTypes.singleOrNull { it.id == type }?.name
                 val dateFrom = Date.fromY_m_d(teacherAbsence.getString("DateFrom"))
                 val dateTo = Date.fromY_m_d(teacherAbsence.getString("DateTo"))
-                val timeFrom = teacherAbsence.getString("TimeFrom")?.let { Time.fromH_m_s(it) }
-                val timeTo = teacherAbsence.getString("TimeTo")?.let { Time.fromH_m_s(it) }
+                val timeFrom = null
+                val timeTo = null
```
Comment (kuba2k2) [inline app/src/main/java/pl/szczodrzynski/edziennik/data/api/edziennik/librus/data/api/LibrusApiTeacherFreeDays.kt:44]: Inline these variables. It's not needed to create a variable for a constant value, so just put them in the constructor of the object below. Also, `"-1".toLong()` is just `-1L`.

**Review by kuba2k2 — CHANGES_REQUESTED** [review]

File: app/src/main/java/pl/szczodrzynski/edziennik/data/api/edziennik/librus/data/api/LibrusApiTeacherFreeDays.kt (hunk 3 above)
Comment (BxOxSxS) [reply app/src/main/java/pl/szczodrzynski/edziennik/data/api/edziennik/librus/data/api/LibrusApiTeacherFreeDays.kt:44]: Im not sure what do you mean, something like this?
```
teacherAbsences?.forEach { teacherAbsence ->
                val id = teacherAbsence.getLong("Id") ?: return@forEach
                val teacherId = teacherAbsence.getJsonObject("Teacher")?.getLong("Id")
                        ?: return@forEach
                val name = data.teacherAbsenceTypes.singleOrNull { it.id == -1L }?.name
                val dateFrom = Date.fromY_m_d(teacherAbsence.getString("DateFrom"))
                val dateTo = Date.fromY_m_d(teacherAbsence.getString("DateTo"))

                val teacherAbsenceObject = TeacherAbsence(
                        profileId = profileId,
                        id = id,
                        type = -1L,
                        name = name,
                        dateFrom = dateFrom,
                        dateTo = dateTo,
                        timeFrom = null,
                        timeTo = null,
                        teacherId = teacherId
                )
```

Comment (kuba2k2) [reply app/src/main/java/pl/szczodrzynski/edziennik/data/api/edziennik/librus/data/api/LibrusApiTeacherFreeDays.kt:44]: yes

File: app/src/main/java/pl/szczodrzynski/edziennik/data/api/edziennik/librus/data/api/LibrusApiTeacherFreeDays.kt [hunk 4]
```diff
@@ -36,22 +36,19 @@ class LibrusApiTeacherFreeDays(override val data: DataLibrus,
                 val id = teacherAbsence.getLong("Id") ?: return@forEach
                 val teacherId = teacherAbsence.getJsonObject("Teacher")?.getLong("Id")
                         ?: return@forEach
-                val type = teacherAbsence.getJsonObject("Type").getLong("Id") ?: return@forEach
-                val name = data.teacherAbsenceTypes.singleOrNull { it.id == type }?.name
+                val name = data.teacherAbsenceTypes.singleOrNull { it.id == -1L }?.name
```
Comment (kuba2k2) [inline app/src/main/java/pl/szczodrzynski/edziennik/data/api/edziennik/librus/data/api/LibrusApiTeacherFreeDays.kt:39]: Ah, forgot about this. As there's no type by that ID anyway, maybe a simple `null` or `""` will be enough.

**Review by kuba2k2 — CHANGES_REQUESTED** [review]

**Review by kuba2k2 — APPROVED** [review]
